### PR TITLE
feat: stat_summary_bin continuous x binned y summary (#817)

### LINKED ISSUE #817: parity: stat_summary_bin, stat_summary_2d, stat_summary_hex — binned summaries
### Request kind
Grammar, stat, geom, scale, coordinate, or PortableSpec

### Problem to solve
Beyond raw counts, ggplot2 summarizes a third variable inside bins: `stat_summary_bin` (1D), `stat_summary_2d`, `stat_summary_hex`. ggsvelte has `summary` for discrete x groups only (errorbar path).

### Scope (missing)
- **stats:** `summary_bin`, `summary_2d`, `summary_hex` (alias summary2d)
- Soft dependencies: `bin` extensions, `bin_2d`, `bin_hex` issues
- Reuse `summary` fun registry (mean/median/…)

### Desired outcome
```ts
{ geom: "line", stat: "summary_bin", mapping: { x: "x", y: "y" }, params: { fun: "mean", bins: 30 } }
{ geom: "tile", stat: "summary_2d", mapping: { x: "x", y: "y", z: "z" }, params: { fun: "mean" } }
{ geom: "hex", stat: "summary_hex", mapping: { x: "x", y: "y", z: "z" } }
```

### Semantic and interaction contract
z/y summary per bin; empty-bin policy; weight channel if already supported elsewhere.

### Portability and rendering
Via tile/hex/line/point geoms.

### Alternatives considered
- Pre-aggregate bins in user SQL/data frame.
- Only count-based bin2d/hex.

### Related Discussion
not discussed.

---

### Clean-room constraint
Implement by matching **behavior and public contracts** against ggplot2 docs/fixtures — do **not** copy ggplot2 R source. Prefer R-fixture tests under `spikes/pure` / package tests where the project already does so for count/bin/boxplot.

### ggsvelte authoring surface (required for each new geom/stat)
1. Add to `KNOWN_GEOMS` / `KNOWN_STATS` + `GEOM_DEFAULTS` in `packages/spec/src/schema-catalog.ts`
2. Layer schema branch in `packages/spec/src/schema-declarations.ts` (+ validate/normalize as needed)
3. Core pipeline: stats under `packages/core/src/stats/`, geometry dispatch / mark batches
4. Svelte `Geom*` child + builder sugar (`.geomX()`) when other geoms have them
5. Example under `examples/` + focused unit tests; VR only if visual contract is load-bearing

### Inventory source
Local inventory against ggplot2 `4.0.3.9000` @ `6870419` and ggsvelte `KNOWN_*` catalogs (worktree slot-3). Not every ggplot2 argument needs v1 parity — document intentional subsetting in the PR.

### PR DESCRIPTION
**Pull request by ljodea**
## Summary

Implements **stat_summary_bin** (#817 v1) — clean-room 1D binned y summaries:

```ts
gg(data, aes({ x: "x", y: "y" }))
  .geomPoint({ stat: "summary_bin", binwidth: 1 })
  .geomErrorbar({ stat: "summary_bin", bins: 10, fun: "mean" })
  .geomLine({ stat: "summary_bin", binwidth: 0.5 })
  .spec();
```

### Behavior
| Concern | Result |
|---------|--------|
| Math | Shared breaks (`bin-breaks`) + `summarizeValues` (mean_se default) |
| Empty bins | Omitted (unlike `stat_bin`) |
| Output | `x` center, `xmin`/`xmax`, `y`/`ymin`/`ymax` |
| Lineage | Bin-edge membership (like `bin`) |
| Surfaces | point, line, errorbar |

### Claude plan review
**APPROVE WITH CHANGES** — applied: bin-shaped lineage, scale-space binning (no y re-transform), PointLayer own union (not IdentityOrUniqueStat), shared summary math extract, facet binRange, weight-unsupported, dual STAT registries.

### Intentional subset
- No `summary_2d` / `summary_hex` (need bin_2d/hex)
- No `aes.weight`
- No empty-bin keep mode

## Test plan
- [x] `bun test packages/core/tests/stats-summary-bin.test.ts` (7)
- [x] `bun test packages/core/tests/pipeline-m2/summary-bin.test.ts` (6)
- [x] summary R-fixture parity still green
- [x] schema emit + artifact
- [ ] CI green

Closes #817

---

<a href="https://app.devin.ai/review/ljodea/ggsvelte/pull/901" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: packages/core/src/pipeline/candidate-construction/represented-rows.ts [hunk 1]
```diff
@@ -102,9 +102,10 @@ export function filterRepresentedSourceRows(input: {
     aggregateXField !== null &&
     outputX !== null &&
     (stat === "count" || stat === "summary" || stat === "boxplot");
-  const needsBin = stat === "bin" && aggregateXField !== null;
+  const needsBin = (stat === "bin" || stat === "summary_bin") && aggregateXField !== null;
   const needsY =
-    (stat === "smooth" || stat === "summary" || stat === "boxplot") && aggregateYField !== null;
+    (stat === "smooth" || stat === "summary" || stat === "summary_bin" || stat === "boxplot") &&
+    aggregateYField !== null;
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/pipeline/candidate-construction/represented-rows.ts]: Fixed in a7189c1e — excluded `summary_bin` from `needsY` so both the indexed bin path and the fallback use **bin-edge membership only** (finite-x, like `stat_bin`). Non-finite y is still dropped when computing the summary value at stat time; it no longer diverges for lineage/tooltips.

@devin-ai-integration

File: packages/core/src/pipeline/candidate-construction/identity-index.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export function buildCandidateIdentityIndex(
           appendSourceRowByGroupKey(sourceRowsByGroupY, key, sourceRow);
         }
       }
-      if (stat === "bin") {
+      if (stat === "bin" || stat === "summary_bin") {
         buildBinLineageBuckets({
           frame,
           panelIndex,
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/pipeline/candidate-construction/identity-index.ts:113]: Valid — `summary_bin` omits empty bins so first/last *rendered* bin is not necessarily bin 0 / final break. Tracked as #905 for a proper fix (gate outer-edge inclusivity on true break extents, plus a pipeline test). Not blocking the #901 land; lineage edge case only.

@devin-ai-integration
